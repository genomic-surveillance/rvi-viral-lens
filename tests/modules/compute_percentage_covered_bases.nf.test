nextflow_process {

    name "Test Process compute_percentage_covered_bases"
    script "modules/compute_percentage_covered_bases.nf"
    process "compute_percentage_covered_bases"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                meta = [
                    id:"test.9000000",
                    sample_id:"test",
                    taxid:9000000
                ]
                test_loc="${projectDir}/tests/test_data"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    params.meta,
                    file("${params.test_loc}/test_consensuses/test.9000000.fa"),
                    file("${params.test_loc}/test_references/9000000.fa")
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
