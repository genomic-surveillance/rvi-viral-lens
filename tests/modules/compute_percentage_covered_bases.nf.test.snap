{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        "99.61993243243244|2369|10|\n"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-02-29T09:00:27.32551"
    }
}